src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,273 @@
+        TransportService transportService,
+        SearchFeatureDao searchFeatureDao
+    ) {
+        super(ValidateAnomalyDetectorAction.NAME, transportService, actionFilters, ValidateAnomalyDetectorRequest::new);
+        this.client = client;
+        this.clusterService = clusterService;
+        this.xContentRegistry = xContentRegistry;
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        this.searchFeatureDao = searchFeatureDao;
+    }
+
+    @Override
+    protected void doExecute(Task task, ValidateAnomalyDetectorRequest request, ActionListener<ValidateAnomalyDetectorResponse> listener) {
+        User user = getUserContext(client);
+        AnomalyDetector anomalyDetector = request.getDetector();
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            resolveUserAndExecute(user, listener, detector -> validateExecute(request, user, context, listener));
+        } catch (Exception e) {
+            logger.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void resolveUserAndExecute(
+        User requestedUser,
+        ActionListener<ValidateAnomalyDetectorResponse> listener,
+        Consumer<AnomalyDetector> function
+    ) {
+        try {
+            // Check if user has backend roles
+            // When filter by is enabled, block users validating detectors who do not have backend roles.
+            if (filterByEnabled && !checkFilterByBackendRoles(requestedUser, listener)) {
+                return;
+            }
+            // Validate Detector
+            function.accept(null);
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    private void validateExecute(
+        ValidateAnomalyDetectorRequest request,
+        User user,
+        ThreadContext.StoredContext storedContext,
+        ActionListener<ValidateAnomalyDetectorResponse> listener
+    ) {
+        storedContext.restore();
+        AnomalyDetector detector = request.getDetector();
+        ActionListener<ValidateAnomalyDetectorResponse> validateListener = ActionListener.wrap(response -> {
+            logger.debug("Result of validation process " + response);
+            // forcing response to be empty
+            listener.onResponse(new ValidateAnomalyDetectorResponse((DetectorValidationIssue) null));
+        }, exception -> {
+            if (exception instanceof ADValidationException) {
+                // ADValidationException is converted as validation issues returned as response to user
+                DetectorValidationIssue issue = parseADValidationException((ADValidationException) exception);
+                listener.onResponse(new ValidateAnomalyDetectorResponse(issue));
```

Comment:
why call `onResponse` here instead of `onFailure`?

Reply:
^ Nvm, I think I understand - after validating, if any errors found, will propagate the issues and fixes via exceptions, which are then parsed into DetectorValidationIssues and a response is built from that. If the exception is caused by something else (illegal argument, etc.), it is considered a failure

To me, seems kind of weird to "wrap" the response via an exception then "unwrap"/parse it here. Is the limitation because it is implementing the abstract handler which originally was throwing exceptions because of a failed attempt to actually update/create the detector?